<a href="https://colab.research.google.com/github/larregui/IST718-Project/blob/main/IST718_FinalProject_2020.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Title: Out of This World
# Author: Laura & Tiffannie
# Purpose: Final Project for IST 718 Big Data Analytics
# Professor: Jon Fox


### Import Libraries



In [2]:
import pandas as pd #data frame operations
import numpy as np #arrays and math functions
from scipy.stats import uniform #for training and test splits
import statsmodels.api as sm  # statistical models (including regression)
import statsmodels.formula.api as smf  # R-like model specification
import matplotlib.pyplot as plt #2D plotting
import seaborn as sns #seaborn for plotting
from sklearn.linear_model import LinearRegression

/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


### Import Dataset and Cleanup

In [4]:
# read in UFO data
ufo = pd.read_csv("ufo.csv")
# shape and data types of the data
print("What are the dimensions of the dataset?\n",ufo.shape)
print("What are the datatypes for each variable?\n",ufo.dtypes)
# compute descriptive statistics 
print("Descriptive Statistics\n",ufo.describe())
# % of missing.
for col in ufo.columns:
    pct_missing = np.mean(ufo[col].isnull())
    print("Are there any null values?\n",'{} - {}%'.format(col, round(pct_missing*100)))

What are the dimensions of the dataset?
 (80332, 11)
What are the datatypes for each variable?
 datetime                 object
city                     object
state                    object
country                  object
shape                    object
duration (seconds)       object
duration (hours/min)     object
comments                 object
date posted              object
latitude                 object
longitude               float64
dtype: object
Descriptive Statistics
          longitude 
count  80332.000000
mean     -86.772885
std       39.697205
min     -176.658056
25%     -112.073333
50%      -87.903611
75%      -78.755000
max      178.441900
Are there any null values?
 datetime - 0.0%
Are there any null values?
 city - 0.0%
Are there any null values?
 state - 7.0%
Are there any null values?
 country - 12.0%
Are there any null values?
 shape - 2.0%
Are there any null values?
 duration (seconds) - 0.0%
Are there any null values?
 duration (hours/min) - 0.0%
Are there any 

/usr/local/lib/python3.6/dist-packages/IPython/core/interactiveshell.py:2718: DtypeWarning: Columns (5,9) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


The dataset has 11 variables and 80,332 records. 
The following columns have missing values:


*   state - 7%
*   Country - 12%
*   Shape - 2%



In [17]:
print(ufo.country.unique())
print(len(ufo.country),"Contries")
# The datase includes records from US, GB, CA, AU, and DE.
# Should we study one country?
nufo = ufo[ufo.country == 'us']
#change attributes data type as needed
#nufo['duration (seconds)'].astype('int32')

['us' nan 'gb' 'ca' 'au' 'de']
80332 Contries


ValueError: ignored

### Data Visualizations

In [10]:
nufo

datetime            city  ...    latitude  longitude 
37571  4/18/2005 00:00       anchorage  ...  61.2180556 -149.900278
45854   5/6/1999 13:30       anchorage  ...  61.2180556 -149.900278
3745   10/2/2012 21:30       anchorage  ...  61.2180556 -149.900278
28656  2/26/2005 05:40           ester  ...  64.8472222 -148.014444
45832   5/5/2014 16:00       anchorage  ...  61.2180556 -149.900278
...                ...             ...  ...         ...         ...
6075   10/3/2003 20:51           clark  ...  44.8994444 -109.150278
64609  8/11/2008 20:15        gillette  ...  44.2911111 -105.501667
26506  2/14/2013 06:00  point of rocks  ...  41.6802778 -108.785000
41209  5/10/2002 19:00        gillette  ...  44.2911111 -105.501667
62769   7/5/1973 01:00          powell  ...  44.7538889 -108.756667

[65114 rows x 11 columns]

In [11]:
#Boxplot By State
nufo.sort_values('state', inplace=True, ascending=True)
#bp = sns.boxplot(x="state",
                         y="duration (seconds)",
                         data=nufo,
                         color = "green")

#add title, xlabel and y label
plt.title('UFO Sighting Duration by State', fontsize = 18)
plt.xlabel('state')
plt.ylabel('duration (seconds)')
#rotate names to reduce overlap
bp.set_xticklabels(bp.get_xticklabels(),rotation=45)
plt.show()

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


TypeError: ignored

### Build a training and testing dataframes

In [ ]:
np.random.seed(1234)
nufo['runiform'] = uniform.rvs(loc = 0, scale = 1, size = len(df))
train = nufo[nufo['runiform'] >= 0.33]
test = nufo[nufo['runiform'] < 0.33]

### Initial data analysis


### Models


### Questions